lib/mayaUsd/render/vp2RenderDelegate/points.cpp

```diff
@@ -0,0 +1,1071 @@
+                    MProfilingScope profilingScope(
+                        HdVP2RenderDelegate::sProfilerCategory,
+                        MProfiler::kColorC_L2,
+                        rprimId.asChar(),
+                        "CommitPositions");
+
+                    positionsBuffer->commit(bufferData);
+                });
+        }
+    }
+
+    _SyncSharedData(_sharedData, delegate, dirtyBits, reprToken, id, _reprs);
+
+    *dirtyBits = HdChangeTracker::Clean;
+
+    // Draw item update is controlled by its own dirty bits.
+    _UpdateRepr(delegate, reprToken);
+}
+
+/*! \brief  Update the draw item
+
+    This call happens on worker threads and results of the change are collected
+    in MayaUsdCommitState and enqueued for Commit on main-thread using CommitTasks
+*/
+void HdVP2Points::_UpdateDrawItem(
+    HdSceneDelegate*        sceneDelegate,
+    HdVP2DrawItem*          drawItem,
+    HdPointsReprDesc const& desc)
+{
+    MHWRender::MRenderItem* renderItem = drawItem->GetRenderItem();
+    if (ARCH_UNLIKELY(!renderItem)) {
+        return;
+    }
+
+    HdDirtyBits itemDirtyBits = drawItem->GetDirtyBits();
+
+    MayaUsdCommitState             stateToCommit(drawItem->GetRenderItemData());
+    HdVP2DrawItem::RenderItemData& drawItemData = stateToCommit._renderItemData;
+
+    const SdfPath& id = GetId();
+
+    auto* const          param = static_cast<HdVP2RenderParam*>(_delegate->GetRenderParam());
+    ProxyRenderDelegate& drawScene = param->GetDrawScene();
+
+    const HdRenderIndex& renderIndex = sceneDelegate->GetRenderIndex();
+
+    const auto& primvarSourceMap = _pointsSharedData._primvarSourceMap;
+
+    const MHWRender::MGeometry::DrawMode drawMode = renderItem->drawMode();
+
+    // The bounding box item uses a globally-shared geometry data therefore it
+    // doesn't need to extract index data from topology. Points use non-indexed
+    // draw.
+    const bool isBoundingBoxItem = (drawMode == MHWRender::MGeometry::kBoundingBox);
+
+#ifdef MAYA_NEW_POINT_SNAPPING_SUPPORT
+    constexpr bool isPointSnappingItem = false;
+#else
+    constexpr bool isPointSnappingItem = true;
+#endif
+
+    if (desc.geomStyle == HdPointsGeomStylePoints) {
+        // Prepare normals buffer.
+        if (itemDirtyBits & (HdChangeTracker::DirtyNormals | HdChangeTracker::DirtyDisplayStyle)) {
+            VtVec3fArray normals;
+
+            const auto it = primvarSourceMap.find(HdTokens->normals);
+            if (it != primvarSourceMap.end()) {
+                const VtValue& value = it->second.data;
+                if (ARCH_LIKELY(value.IsHolding<VtVec3fArray>())) {
+                    normals = value.UncheckedGet<VtVec3fArray>();
+                }
+            }
+
+            // Using a zero vector to indicate requirement of camera-facing
+            // normals when there is no authored normals.
+            if (normals.empty()) {
+                normals.push_back(GfVec3f(0.0f, 0.0f, 0.0f));
```

Comment:
What if the user has authored normals which are the zero vector for some reason?